openmdao/devtools/xdsm_viewer/xdsm_writer.py

```diff
@@ -982,14 +993,33 @@ def _accumulate_connections(conns):
     return conns_new
 
 
-def _collect_connections(variables, recurse):
+def _collect_connections(variables, recurse, model_path=None):
     conv_vars = [_convert_name(v, recurse) for v in variables]
     connections = dict()
     for conv_var in conv_vars:
-        connections.setdefault(conv_var['path'], []).append(conv_var['var'])
+        path = _make_rel_path(conv_var['path'], model_path=model_path)
+        connections.setdefault(path, []).append(conv_var['var'])
     return connections
 
 
+def _get_path(name, sep='.'):
+    # Returns path until the last separator in the name
+    return name.rsplit(sep, 1)[0]
+
+
+def _make_rel_path(full_path, model_path):
+    # Path will be cut from this character. Length of model path + separator after it.
+    # If path does not contain the model path, the full path will be returned.
+    if model_path is not None:
+        first_char = len(model_path) + 1
+        if full_path.startswith(model_path):
```

Comment:
this could give false matches unless you check for startswith(model_path + '.') instead of startswith(model_path)